modules/adding-ibmz-lpar-agent.adoc

```diff
@@ -0,0 +1,92 @@
+<3> For installations on DASD-type disks, use `rd.dasd` to specify the DASD where RedHat Enterprise Linux CoreOS (RHCOS) is to be installed. For installations on FCP-type disks, use `rd.zfcp=<adapter>,<wwpn>,<lun>` to specify the FCP disk where RHCOS is to be installed.
+<4> Specify this parameter when you use an Open Systems Adapter (OSA) or HiperSockets.
+
+. Generate the `.ins` and `initrd.img.addrsize` files.
++
+The `.ins` file is a special file that includes installation data and is present on the FTP server. It can be accessed from the HMC system.
+This file contains details such as mapping of the location of installation data on the disk or FTP server, the memory locations where the data is to be copied.
++
+[NOTE]
+====
+The `.ins` and `initrd.img.addrsize` files are not automatically generated as part of boot-artifacts from the installer. You must manually generate these files.
+====
+
+.. Save the following script to a file, such as `generate-files.py`:
++
+.`generate-files.py` file
+[source,python]
+----
+# The following commands retrieve the size of the `kernel` and `initrd`:
+KERNEL_IMG_PATH='./kernel.img'
+INITRD_IMG_PATH='./initrd.img'
+CMDLINE_PATH='./generic.prm'
+kernel_size=$(stat -c%s $KERNEL_IMG_PATH)
+initrd_size=$(stat -c%s $INITRD_IMG_PATH)
+# The following command rounds the `kernel` size up to the next megabytes (MB) boundary.
+# This value is the starting address of `initrd.img`.
+offset=$(( (kernel_size + 1048575) / 1048576 * 1048576 ))
+INITRD_IMG_NAME=$(echo $INITRD_IMG_PATH | rev | cut -d '/' -f 1 | rev)
+# The following commands create the kernel binary patch file that contains the `initrd` address and size:
+KERNEL_OFFSET=0x00000000
+KERNEL_CMDLINE_OFFSET=0x00010480
+INITRD_ADDR_SIZE_OFFSET=0x00010408
+OFFSET_HEX=$(printf '0x%08x\n' $offset)
+# The following command converts the address and size to binary format:
+printf "$(printf '%016x\n' $initrd_size)" | xxd -r -p > temp_size.bin
+# The following command concatenates the address and size binaries:
+cat temp_address.bin temp_size.bin > "$INITRD_IMG_NAME.addrsize"
+# The following command deletes temporary files:
+rm -rf temp_address.bin temp_size.bin
+# The following commands create the `.ins` file.
+# The file is based on the paths of the `kernel.img`, `initrd.img`, `initrd.img.addrsize`, and `cmdline` files and the memory locations where the data is to be copied.
+$KERNEL_IMG_PATH $KERNEL_OFFSET
+$INITRD_IMG_PATH $OFFSET_HEX
+$INITRD_IMG_NAME.addrsize $INITRD_ADDR_SIZE_OFFSET
+$CMDLINE_PATH $KERNEL_CMDLINE_OFFSET
+----
+
+.. Execute the script by running the following command:
++
+[source,terminal]
+----
+$ python3 generate-file.py
+----
+
+. Transfer the `initrd`, `kernel`, `generic.ins`, and `initrd.img.addrsize` parameter files to the file server. For more information, see link:https://www.ibm.com/docs/en/linux-on-systems?topic=bl-booting-linux-in-lpar-mode[Booting Linux in LPAR mode].
```

Comment:
```suggestion
. Transfer the `initrd`, `kernel`, `generic.ins`, and `initrd.img.addrsize` parameter files to the file server. For more information, see link:https://www.ibm.com/docs/en/linux-on-systems?topic=bl-booting-linux-in-lpar-mode[Booting Linux in LPAR mode (IBM documentation].
```

This aspect is importent to let a customer know that they will need to leave our customer documentation to visit IBM's documentation portal, which luckily does not currently require a login operation.